cmd/cosign/webhook/main.go

```diff
@@ -17,99 +17,74 @@ package main
 
 import (
 	"context"
-	goflag "flag"
-	"net"
-	"net/http"
-	"os"
-	"strconv"
+	"flag"
 
-	flag "github.com/spf13/pflag"
 	appsv1 "k8s.io/api/apps/v1"
 	batchv1 "k8s.io/api/batch/v1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/klog/v2"
-	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/healthz"
-	"sigs.k8s.io/controller-runtime/pkg/log/zap"
-
-	"github.com/sigstore/cosign/pkg/cosign/kubernetes/webhook"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/configmap"
+	"knative.dev/pkg/controller"
+	"knative.dev/pkg/injection/sharedmain"
+	"knative.dev/pkg/signals"
+	"knative.dev/pkg/webhook"
+	"knative.dev/pkg/webhook/certificates"
+	"knative.dev/pkg/webhook/resourcesemantics"
+	"knative.dev/pkg/webhook/resourcesemantics/validation"
+
+	cwebhook "github.com/sigstore/cosign/pkg/cosign/kubernetes/webhook"
 )
 
-func main() {
-	ctrl.SetLogger(zap.New(func(o *zap.Options) {
-		o.Development = true
-	}))
-
-	var (
-		metricsAddr        = net.ParseIP("127.0.0.1")
-		metricsPort uint16 = 8080
+var secretName = flag.String("secret-name", "", "The name of the secret in the webhook's namespace.")
 
-		bindAddr        = net.ParseIP("0.0.0.0")
-		bindPort uint16 = 8443
-
-		tlsCertDirectory string
-		secretKeyRef     string
+func main() {
+	ctx := webhook.WithOptions(signals.NewContext(), webhook.Options{
+		ServiceName: "webhook",
+		Port:        8443,
```

Comment:
Port should remain configurable.
Metrics port and other settings would need to be configurable as well.

Reply:
I take that back.  The configmap controls a nested configuration.  The prometheus port is configurable through environment variables: https://github.com/knative/pkg/blob/50410e0b833abc1a464d334b9e52e2c873dafcd4/metrics/config.go#L55-L56